elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java

```diff
@@ -263,6 +257,12 @@ String scriptedFieldPrefix() {
     // cleanup query. remove empty AGGREGATIONS element (if empty)
     emptyAggRemover.accept(query);
 
+    // This must be set to true or else in 7.X and 6/7 mixed clusters
+    // will return lower bounded count values instead of an accurate count.
+    if (groupBy.isEmpty() && ElasticsearchVersion.ES6.compareTo(version) <= 0) {
```

Comment:
I wonder if comparing enum ordinals is a little error-prone ? 
Not sure there is a better option without larger code changes (maybe instead of enum have `int elasticVersionMajor`).

Reply:
done